iroh-base/src/key.rs

```diff
@@ -353,14 +348,14 @@ fn decode_base32_hex(s: &str) -> Result<[u8; 32], KeyParsingError> {
         let input = s.to_ascii_uppercase();
         let input = input.as_bytes();
         if data_encoding::BASE32_NOPAD.decode_len(input.len())? != bytes.len() {
-            return Err(DecodeInvalidLengthSnafu.build());
+            return Err(KeyParsingError::decode_invalid_length());
```

Comment:
I guess this means rust-analyzer "find-all-references" on `KeyParsingError::DecodeInvalidLength` is not going to find anything?

That's one of the main usability issues I have with snafu.